beeline/src/java/org/apache/hive/beeline/schematool/HiveSchemaTool.java

```diff
@@ -112,6 +121,40 @@ protected void execSql(String sqlScriptFile) throws IOException {
     }
   }
 
+  void replaceLocationForProtoLogTables(String sqlScriptFile) throws IOException {
+    TezConfiguration tezConf = new TezConfiguration(true);
+    String hiveProtoBaseDir = HiveConf.getVar(conf, HiveConf.ConfVars.HIVE_PROTO_EVENTS_BASE_PATH);
+    String tezProtoBaseDir = tezConf.get(TezConfiguration.TEZ_HISTORY_LOGGING_PROTO_BASE_DIR);
+
+    if (isEmpty(hiveProtoBaseDir)) {
+      throw new IOException("Hive conf variable hive.hook.proto.base-directory is not set for creating protologging tables");
+    }
+    if (isEmpty(tezProtoBaseDir)) {
+      throw new IOException("Tez conf variable tez.history.logging.proto-base-dir is not set for creating protologging tables");
+    }
+
+    Map<String, String> replacements = new HashMap<>();
+    replacements.put("_REPLACE_WITH_QUERY_DATA_LOCATION_", "hdfs:///" + hiveProtoBaseDir);
```

Comment:
I would say we may avoid hardcoding the prefix, i.e., following the logic using HIVE_PROTO_EVENTS_BASE_PATH.
https://github.com/apache/hive/blob/rel/release-4.0.0-beta-1/ql/src/java/org/apache/tez/dag/history/logging/proto/DatePartitionedLogger.java#L67

As far as I tested, `hive.hook.proto.base-directory` can accept `hdfs://path/to/proto` or `s3a://bucket/path/to/proto`.

Reply:
Sorry for the late update as I was away due to personal issues. I have updated the files by working on your suggestions.